internal/compiler/diagnostics.rs

```diff
@@ -520,3 +520,85 @@ impl BuildDiagnostics {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
nitpick:  there is no need to use a `tests` module.
Usually we only need these module if more functions or item needs to be added to support the tests, but this is not the case here.

Reply:
They are needed if you add helper functions or helper struct in it. Otherwise they are not needed.
What is "the crowd"? I personally prefer putting the test right after the function they test.